parsing/builtin_attributes.ml

```diff
@@ -188,3 +188,22 @@ let explicit_arity =
       | ({txt="ocaml.explicit_arity"|"explicit_arity"; _}, _) -> true
       | _ -> false
     )
+
+(* The "ocaml.boxed (default)" and "ocaml.unboxed (default)"
+   attributes cannot be input by the user, they are added by the
+   compiler when applying the default setting. This is done to record
+   in the .cmi the default used by the compiler when compiling the
+   source file because the default can change between compiler
+   invocations. *)
+
```

Comment:
I second that. There is too much stringly typed stuff in the compiler already.

It might be worth having some support for ints containing sets of flags at some point, as I sometimes suspect that people don't like adding a whole word to a record just for a boolean and that pushes them to encode things in strings or attributes.